pkg/mcs/scheduling/server/cluster.go

```diff
@@ -224,6 +233,22 @@ func (c *Cluster) updateScheduler() {
 			// This is triggered by the watcher when the schedulers are updated.
 		}
 
+		if !c.running.Load() {
+			select {
+			case <-c.ctx.Done():
+				log.Info("cluster is closing, stop listening the schedulers updating notifier")
+				return
+			case <-ticker.C:
+				// retry
+				select {
+				case notifier <- struct{}{}:
+				// If the channel is not empty, it means the check is triggered.
+				default:
+				}
```

Comment:
What about warping a `trySend` function to reuse the code?

Reply:
sounds good